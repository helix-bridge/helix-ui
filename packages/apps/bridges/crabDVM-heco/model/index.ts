export * from './bridge';
export * from './tx';
