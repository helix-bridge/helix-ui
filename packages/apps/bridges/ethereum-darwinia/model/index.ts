export * from './bridge';
export * from './cross-chain';
export * from './deposit';
export * from './record';
