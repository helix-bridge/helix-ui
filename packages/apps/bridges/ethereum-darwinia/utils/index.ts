export * from './tx';
export * from './fee';
