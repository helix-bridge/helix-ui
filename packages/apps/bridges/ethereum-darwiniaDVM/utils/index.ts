export * from './register';
export * from './tx';
