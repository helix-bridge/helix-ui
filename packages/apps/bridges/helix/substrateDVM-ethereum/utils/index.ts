export * from './tx';
export * from './fee';
export * from './dailyLimit';
