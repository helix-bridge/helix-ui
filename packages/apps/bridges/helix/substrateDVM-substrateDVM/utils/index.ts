export * from './bridge';
export * from './bridge-inner';
