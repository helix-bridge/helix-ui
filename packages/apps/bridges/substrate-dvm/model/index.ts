export * from './bridge';
export * from './cross-chain';
export * from './record';
