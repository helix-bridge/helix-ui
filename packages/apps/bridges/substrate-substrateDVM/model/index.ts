export * from './cross-chain';
export * from './bridge';
export * from './record';
