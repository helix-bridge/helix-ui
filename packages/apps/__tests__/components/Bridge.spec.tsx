/// <reference types="jest" />

import { create } from 'react-test-renderer';
import { BridgeCategory, HelixHistoryRecord } from 'shared/model';
import { Bridge } from '../../components/transaction/Bridge';
import { getOriginChainConfig } from '../../utils/network';
import s2sv2 from '../fixture/darwin-dvm-crab-dvm.json';

jest.mock('next/router', () => ({
  useRouter() {
    return {
      route: '/',
      pathname: '',
      query: {
        from: 'darwinia',
        to: 'crab-dvm',
      },
      asPath: '',
    };
  },
}));

describe('<Bridge />', () => {
  it('render bridge', () => {
    const record = s2sv2[0] as HelixHistoryRecord;
    const component = create(
      <Bridge
        from={getOriginChainConfig(record.fromChain)}
        to={getOriginChainConfig(record.toChain)}
        category={record.bridge.split('-')[0] as BridgeCategory}
      />
    );

    let tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
