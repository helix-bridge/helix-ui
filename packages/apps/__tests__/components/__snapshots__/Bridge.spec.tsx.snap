// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Bridge /> render bridge 1`] = `
<div
  className="flex flex-wrap md:items-center gap-2 md:gap-16 my-3 md:my-6 relative"
>
  <div
    className="w-36 flex items-center gap-2"
  >
    <span
      className="capitalize"
    >
      Transfer Route
    </span>
  </div>
  <div
    className="flex justify-between items-center gap-2 bg-antDark bg-opacity-25"
  >
    <div
      className="flex items-center gap-2"
    >
      <div>
        Darwinia
      </div>
      <span
        style={
          Object {
            "background": "none",
            "border": 0,
            "boxSizing": "border-box",
            "display": "inline-block",
            "height": "initial",
            "margin": 0,
            "maxWidth": "100%",
            "opacity": 1,
            "overflow": "hidden",
            "padding": 0,
            "position": "relative",
            "width": "initial",
          }
        }
      >
        <span
          style={
            Object {
              "background": "none",
              "border": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": "initial",
              "margin": 0,
              "maxWidth": "100%",
              "opacity": 1,
              "padding": 0,
              "width": "initial",
            }
          }
        >
          <img
            alt=""
            aria-hidden={true}
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2740%27%20height=%2740%27/%3e"
            style={
              Object {
                "background": "none",
                "border": 0,
                "display": "block",
                "height": "initial",
                "margin": 0,
                "maxWidth": "100%",
                "opacity": 1,
                "padding": 0,
                "width": "initial",
              }
            }
          />
        </span>
        <img
          className="rounded-full w-5 md:w-10"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
        <noscript>
          <img
            className="rounded-full w-5 md:w-10"
            data-nimg="intrinsic"
            decoding="async"
            loading="lazy"
            src="/_next/image?url=%2Fimage%2F&w=96&q=75"
            srcSet="/_next/image?url=%2Fimage%2F&w=48&q=75 1x, /_next/image?url=%2Fimage%2F&w=96&q=75 2x"
            style={
              Object {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": undefined,
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": 0,
              }
            }
          />
        </noscript>
      </span>
    </div>
    <svg
      className="icon w-4 h-4"
      onClick={[Function]}
    />
    <div
      className="bg-black p-2 px-4 rounded-3xl"
    >
      <img
        className="w-10 md:w-28"
        src="/image/bridges/helix.png"
      />
    </div>
    <svg
      className="icon w-4 h-4"
      onClick={[Function]}
    />
    <div
      className="flex items-center gap-2"
    >
      <span
        style={
          Object {
            "background": "none",
            "border": 0,
            "boxSizing": "border-box",
            "display": "inline-block",
            "height": "initial",
            "margin": 0,
            "maxWidth": "100%",
            "opacity": 1,
            "overflow": "hidden",
            "padding": 0,
            "position": "relative",
            "width": "initial",
          }
        }
      >
        <span
          style={
            Object {
              "background": "none",
              "border": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": "initial",
              "margin": 0,
              "maxWidth": "100%",
              "opacity": 1,
              "padding": 0,
              "width": "initial",
            }
          }
        >
          <img
            alt=""
            aria-hidden={true}
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2740%27%20height=%2740%27/%3e"
            style={
              Object {
                "background": "none",
                "border": 0,
                "display": "block",
                "height": "initial",
                "margin": 0,
                "maxWidth": "100%",
                "opacity": 1,
                "padding": 0,
                "width": "initial",
              }
            }
          />
        </span>
        <img
          className="rounded-full w-5 md:w-10"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
        <noscript>
          <img
            className="rounded-full w-5 md:w-10"
            data-nimg="intrinsic"
            decoding="async"
            loading="lazy"
            src="/_next/image?url=%2Fimage%2F&w=96&q=75"
            srcSet="/_next/image?url=%2Fimage%2F&w=48&q=75 1x, /_next/image?url=%2Fimage%2F&w=96&q=75 2x"
            style={
              Object {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": undefined,
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": 0,
              }
            }
          />
        </noscript>
      </span>
      <div>
        Crab
      </div>
    </div>
  </div>
</div>
`;
